talk/morelanguage/raii.tex

```diff
@@ -318,9 +320,11 @@
   \end{exampleblock}
 \end{frame}
 
+\end{advanced}
+
 \begin{frame}[fragile]
-  \frametitlecpp[11]{Quiz: shared\_ptr and weak\_ptr}
-  \begin{exampleblockGB}{What is the output of this code?}{https://godbolt.org/z/vM35Y6qEW}{shared\_ptr vs weak\_ptr}
+  \frametitlecpp[11]{Quiz: shared\_ptr in use}
+  \begin{exampleblockGB}{What is the output of this code?}{https://godbolt.org/z/vM35Y6qEW}{shared\_ptr quiz}
```

Comment:
As pointed out in my post-merge feedback to #393, I think we typeset C++ constructs in headers using `\texttt{...}`:
```suggestion
  \frametitlecpp[11]{Quiz: \texttt{shared\_ptr} in use}
  \begin{exampleblockGB}{What is the output of this code?}{https://godbolt.org/z/vM35Y6qEW}{\texttt{shared\_ptr} quiz}
```

Reply:
At best, we are inconsistent:
<img width="268" alt="image" src="https://user-images.githubusercontent.com/16205615/221151429-2bd878c3-4f53-4dd4-accf-4e108f509b04.png">
<img width="378" alt="image" src="https://user-images.githubusercontent.com/16205615/221151496-7a33d87d-1179-4ea7-9188-f090821569c9.png">
<img width="285" alt="image" src="https://user-images.githubusercontent.com/16205615/221151559-e81204f0-0282-4732-a417-743db7d62042.png">
<img width="310" alt="image" src="https://user-images.githubusercontent.com/16205615/221151601-d55ac3e9-9b18-4aa3-9c12-261e7662f528.png">
<img width="393" alt="image" src="https://user-images.githubusercontent.com/16205615/221151667-4810b1f6-0a71-4bb3-960c-f0402249e2aa.png">
<img width="123" alt="image" src="https://user-images.githubusercontent.com/16205615/221151751-1c27d40c-5c5f-4f4d-afcb-f8deab5cd725.png">

I agree it should be discussed. Let's also ask @sponce 

In this specific case, however, (line 327), I'm not even sure that a typewriter font is a good choice. The last argument is what ends up in this list:
<img width="572" alt="image" src="https://user-images.githubusercontent.com/16205615/221152323-a5cb9fbf-9379-4731-95ab-b05a742ae003.png">

Here's a proposal, but feel free to chime in:
| Environment | What to do |
| --- | --- |
| `\frametitlecpp` | `\texttt{ }` |
| Chapter/section title | `\texttt{ }` |
| short chapter/section title | normal font? `\texttt{ }`?? (might break when linking / indexing) |
| Box title | `\mintinline{cpp}{ ... }` |
| Inline text | `\mintinline{cpp}{ ... }` |
| Index of XXX (like image above) | normal font? |

Reply:
Ah ok, so maybe use a normal font for the index then. The section titles may be normal as well to fit on the overview slide. Otherwise, I would use `\texttt{}` whereever we can. And yes, that is currently very inconsistent.

Reply:
I would respect the simple rule "always use \mintinline unless the context does not allow, then go back to \texttt". But basically it matches what @hageboeck proposes, answering no to the 2 questions about normal font.

Reply:
Now done. I'll merge when the CI is green.

PS @sponce @bernhardmgruber:
I checked if `\mintinline{cpp}` works in frame titles. When it finds something to highlight, it gets hard to read.